### ISSUE TITLE
Nested variant can only match the first of the nested variants

### ISSUE THREAD

**Issue by cuberoot** [fabian-hiller/valibot#310]
I have a case where I am nesting variants. I discriminate first based on the `type`, and then for some nodes values, on the `subtype`. However, valibot fails on any `subtype` except the first one in the list.

Given this schema:

```typescript
import { literal, number, object, parse, variant } from 'valibot'

const procedural = variant("subtype", [
  object({
    type: literal("procedural"),
    subtype: literal("cache"),
    depth: number(),
  }),
  object({
    type: literal("procedural"),
    subtype: literal("const"),
    value: number(),
  }),
]);

const layer = variant("type", [
  object({
    type: literal("circle"),
    radius: number(),
  }),
  object({
    type: literal("rect"),
    width: number(),
    height: number(),
  }),
  procedural,
]);
```

Using the `procedural` schema, I can match both of these:

```typescript
parse(procedural, { type: "procedural", subtype: "cache", depth: 20 });
parse(procedural, { type: "procedural", subtype: "const", value: 11 });
```

However, using the wider `layer` schema, I can only match the first one.

```typescript
parse(layer, { type: "procedural", subtype: "cache", depth: 20 });
// This one fails with a "Type error"
parse(layer, { type: "procedural", subtype: "const", value: 11 });
```

**Comment by fabian-hiller** [comment]
I think this behavior is currently intended. When I implemented `variant`, I assumed that no two options would use the same discriminator value. I will look into this and try to find a solution.

**Comment by fabian-hiller** [comment]
This is fixed in v0.25.0
